src/lib/features/integration-events/integration-events-store.ts

```diff
@@ -41,11 +41,21 @@ export class IntegrationEventsStore extends CRUDStore<
                 qb.select('id')
                     .from(this.tableName)
                     .whereRaw(`created_at >= now() - INTERVAL '2 hours'`)
-                    .orderBy('created_at', 'desc')
+                    .orderBy('id', 'desc')
                     .limit(100);
             })
+            .with('latest_per_integration', (qb) => {
+                qb.select(this.db.raw('DISTINCT ON (integration_id) id'))
+                    .from(this.tableName)
+                    .orderBy('integration_id')
+                    .orderBy('id', 'desc');
+            })
             .from(this.tableName)
             .whereNotIn('id', this.db.select('id').from('latest_events'))
+            .whereNotIn(
+                'id',
+                this.db.select('id').from('latest_per_integration'),
```

Comment:
How big this can be? I assume this depends on the response of the comment above

Reply:
This will be as big as the total number of integration configurations in the instance, which I don't expect to be a high number.
Again, maybe this will help: https://github.com/Unleash/unleash/pull/7652/commits/272cd0fb4094cc29f3837e9527adca4b1c37f86d